### ISSUE TITLE
Consider adding support for more storage.k8s.io resources

### ISSUE THREAD

**Issue by matskiv** [loft-sh/vcluster#725]
### Is your feature request related to a problem?

Some applications or controllers may need to get/list resources from some [x.csinodes.storage.k8s.io](http://x.csinodes.storage.k8s.io).
For example cluster autoscaler tries to get [csinodes.storage.k8s.io](http://csinodes.storage.k8s.io) resource. But I am not sure what is the exact impact of it not being able to get any CSINodes.

### Which solution do you suggest?

Optionally sync these resources from the host cluster to vcluster.

### Which alternative solutions exist?

Writing a custom plugin.

### Additional context

*No response*

**Comment by matskiv** [comment]
@Ab-hishek we have not yet decided if we want to implement this, so for now we will leave it unassigned.

**Comment by rohantmp** [comment]
The scheduler uses CSINode and complains if it can't fetch it in some cases: https://github.com/kubernetes/kubernetes/blob/462f41212a4e5ded8974784038f681a08dded1be/pkg/scheduler/framework/plugins/volumebinding/binder.go#L561

Thought it was capacity, but looks like that uses a different object. Not sure exactly what.

**Comment by rohantmp** [comment]
IMO, when the internal scheduler is enabled, we will need to add some additional objects. `CSIDrivers`, `StorageCapacities`, `CSINodes`. I will investigate.

**Comment by rohantmp** [comment]
Looks like we need this: 

https://github.com/kubernetes/kubernetes/blob/462f41212a4e5ded8974784038f681a08dded1be/pkg/scheduler/framework/plugins/nodevolumelimits/csi.go#L95

`getVolumeLimits` here fallsback to getting volume limits from node status, but this may be discontinued in a patch release to v1.18
```
	// TODO: stop getting values from Node object in v1.18
	nodeVolumeLimits := volumeLimits(nodeInfo)
	if csiNode != nil {
		for i := range csiNode.Spec.Drivers {
			d := csiNode.Spec.Drivers[i]
			if d.Allocatable != nil && d.Allocatable.Count != nil {
				// TODO: drop GetCSIAttachLimitKey once we don't get values from Node object (v1.18)
				k := v1.ResourceName(volumeutil.GetCSIAttachLimitKey(d.Name))
				nodeVolumeLimits[k] = int64(*d.Allocatable.Count)
			}
		}
	}
```

**Comment by rohantmp** [comment]
Will be doing as a fix for #773
